src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -2095,6 +2108,24 @@ class ObservationStore(
     }
   }
 
+  private enum class UpdateSpeciesType {
+    Plot,
+    Subzone,
+    Zone,
+    Site;
+
+    fun <ID : Any> getTempZoneCondition(scopeId: ID): Condition =
```

Comment:
Would it make sense to pull the other table-specific conditions from `updateSpeciesTotalsTable` into this enum too?

Alternately, instead of an enum with a `when`, make this an interface with implementations for plot, subzone, zone, and site; then you wouldn't need to do the `scopeId as Whatever` casts since each implementation would use the correct types for its scope.